laravel/app/Console/Commands/InitiateEval.php

```diff
@@ -105,152 +106,114 @@ public function initResidentEvalAttendingPOST($evaluatorID, $evaluateeID)
+                        $totalEvalsSent +=  self::sendEvaluations($pair["residentID"], $pair["residentName"], $medhubFormId, $pair["attendingID"], $pair["attendingName"], $date);
+                    }
                 }
+            } else {
+                Log::debug("Cannot initiate evaluation because no resident ID for " . $pair["residentName"] . " or no attending ID for " . $pair["attendingName"] . " can be found.");
             }
-            // either the resident of the attending werent valid, mark it down -- used for testing only
-            if ($evalSent == 0) {
-                //print($evalID[$i].' '.$evaluateeID.' '.$evaluatorID);
+        }
+        Log::debug("Total Number of Evaluations Succesfully sent: " . $totalEvalsSent);
+    }
+
+    public function getResidentAttendingPairs($date)
+    {
+        $pairs = array();
+        //get residents that have evaluation data in DB on day
+        $residents = EvaluateData::where("date", $date)->pluck("resident_id", "resident");
+        Log::info("Residents in Evaluation Data with date " . $date . ":  " . $residents);
+        // for each resident get their active rotation service and create pairs of residents and attendings that worked together
+        foreach ($residents as $residentName => $residentID) {
+            //get Resident's active rotation service
+            $activeService = 0;
+            $activeRotation = Rotations::where("name", $residentName)->where("Start", "<=", $date)->where("End", ">=", $date)->first();
+            if (is_null($activeRotation)) {
+                Log::info("Could not find an active rotation for Resident " . $residentName . " ID: " . $residentID);
+            } else {
+                $activeService = $activeRotation["Service"];
+            }
+            Log::info("Resident " . $residentName . " has active service id " . $activeService);
+            $attendings = EvaluateData::where("date", $date)->where("resident_id", $residentID)->pluck("attending_id", "attending")->unique();
+            foreach ($attendings as $attendingName => $attendingID) {
+                $overallTime = EvaluateData::where("date", $date)->where("resident_id", $residentID)->where("attending_id", $attendingID)->sum("time_with_attending");
+                Log::info("Resident: " . $residentName . "(" . $residentID . ") Attending: " . $attendingName . "(" . $attendingID . ") Total Time Together: " . $overallTime);
+                $residentAndAttending = array(
+                    "residentName" => $residentName,
+                    "residentID" => $residentID,
+                    "residentActiveService" => $activeService,
+                    "attendingID" => $attendingID,
+                    "attendingName" => $attendingName,
```

Comment:
I am using plucks because the logic made more sense to me - originally I was only plucking ids but plucked names as well for logging purposes only. 
- Get the unique residents in table for date
- Get the unique attendings that a given resident worked with for that date
- Calculate the sum of the diff for the cases they worked together
No other data is really useful in the rows and the queries are easy to follow.

Reply:
I mostly say this because it might allow us to return "true" pairs (a `Resident` and an `Attending`), rather than an array of arrays.

Reply:
I think we might thinking of pairs differently - we send evaluations by for each pair of resident and attending - how every the pair needs to have the total amount of time calculated which I think is quite quick and easy to understand by using the sum query. I consider it to be more like an object (except is stored in array) - Resident/Attending pair - they have their ids, names, and the total time together.  

I did change name so that should make it more clear.

Reply:
A pair is always exactly two items, but the array you're working with here has 5 (residentName, residentID, residentActiveService, attendingID, attendingName).

I'd like to either see this become a true pair (i.e. `[$resident, $attending]`) or to rename things so we don't refer to an array with 5 items as a pair.